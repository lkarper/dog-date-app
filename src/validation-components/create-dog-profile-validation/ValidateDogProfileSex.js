import React, { useEffect } from 'react';

const ValidateDogProfileSex = (props) => {

    const { sex, sexError, setSexError } = props;

    useEffect(() => {
        if (sex) {
            setSexError('');
        } else {
            setSexError(`You must select your dog's sex.`);
        }
    }, [sex, setSexError]);

    if (sex) {
        return <p id='sex-validator'>Sex set to: '{sex}.'</p>;
    }

    return <p id='sex-validator'>{sexError}</p>;
}

export default ValidateDogProfileSex;
