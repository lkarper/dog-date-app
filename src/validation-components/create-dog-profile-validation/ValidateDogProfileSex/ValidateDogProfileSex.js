import React, { useEffect } from 'react';

const ValidateDogProfileSex = (props) => {

    const { sex, sexError, setSexError } = props;

    useEffect(() => {
        if (sex) {
            setSexError('');
        } else {
            setSexError(`You must select your dog's sex.`);
        }
    }, [sex, setSexError]);

    if (sex) {
        return (
            <p 
                className='ValidateDogProfileSex__validator valid'
                id='sex-validator'
            >
                Sex set to: '{sex}.'
            </p>
        );
    }

    return (
        <p 
            className='ValidateDogProfileSex__validator error'
            id='sex-validator'
        >
            {sexError}
        </p>
    );
}

export default ValidateDogProfileSex;
