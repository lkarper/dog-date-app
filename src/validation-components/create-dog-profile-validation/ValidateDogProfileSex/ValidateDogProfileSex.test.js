import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateDogProfileSex from './ValidateDogProfileSex';

describe('ValidateDogProfileSex component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateDogProfileSex />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateDogProfileSex />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
