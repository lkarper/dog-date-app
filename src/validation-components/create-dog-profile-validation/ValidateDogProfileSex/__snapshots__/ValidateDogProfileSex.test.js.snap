// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateDogProfileSex component renders the UI as expected 1`] = `
<ValidateDogProfileSex
  setSexError={[Function]}
  sex=""
  sexError="You must select your dog's sex."
>
  <p
    className="ValidateDogProfileSex__validator error"
    id="sex-validator"
  >
    You must select your dog's sex.
  </p>
</ValidateDogProfileSex>
`;
